firebase_admin/messaging.py

```diff
@@ -299,6 +447,18 @@ def make_topic_management_request(self, tokens, topic, operation):
         else:
             return TopicManagementResponse(resp)
 
+    def _message_data(self, message, dry_run):
+        data = {'message': _MessagingService.encode_message(message)}
+        if dry_run:
+            data['validate_only'] = True
+        return data
+
+    def _postproc(self, resp, body):
+        if resp.status == 200:
+            return json.loads(body)
+        else:
+            raise Exception('unexpected response')
```

Comment:
I used this in my hack. But we need to raise something more detailed here. I think we need to parse the error response body, and construct a `messaging.ApiCallError`.

Reply:
Is there a situation where FCM will return a non-200 or non-error response code? I'm trying to figure out what to do here. A `messaging.ApiCallError` takes an error, which we really don't have in this case.

Reply:
Technically it can be any HTTP response. But in practice FCM mostly returns 200 or 4xx/5xx error codes. We can treat anything that is not 200 as an error. 

We need to somehow call the error parsing logic in `_handle_fcm_error()` on this, and get an `ApiCallError` out of it. May be refactor the parsing logic into a new helper function:

```
if resp.status != 200:
  code, message = _parse_error_response(response)
  raise ApiCallError(code, message) # last argument is optional  
```